toolchain/check/cpp/import.cpp

```diff
@@ -1342,21 +1347,28 @@ static auto MakeImplicitParamPatternsBlockId(
 // TODO: Consider refactoring to extract and reuse more logic from
 // `HandleAnyBindingPattern()`.
 static auto MakeParamPatternsBlockId(Context& context, SemIR::LocId loc_id,
-                                     const clang::FunctionDecl& clang_decl)
-    -> SemIR::InstBlockId {
-  if (clang_decl.parameters().empty()) {
+                                     const clang::FunctionDecl& clang_decl,
+                                     int num_params) -> SemIR::InstBlockId {
+  if (clang_decl.parameters().empty() || !num_params) {
```

Comment:
Minor, but I don't think as much looking at an integer comparison as looking at an implicit cast of an integer to a bool.

```suggestion
  if (clang_decl.parameters().empty() || num_params == 0) {
```